plugins/stack-overflow/src/plugin.ts

```diff
@@ -28,6 +31,18 @@ import { StackOverflowQuestionsContentProps } from './types';
  */
 export const stackOverflowPlugin = createPlugin({
   id: 'stack-overflow',
+  apis: [
+    createApiFactory({
+      api: stackOverflowApiRef,
+      deps: { configApi: configApiRef },
+      factory: ({ configApi }) =>
+        new StackOverflowClient({
+          baseUrl:
+            configApi.getOptionalString('stackoverflow.baseUrl') ||
```

Comment:
what do you think about having a static fromConfig method on the `StackOverflowClient` class instead which reads directly from the config? following a similar pattern we use here for example https://github.com/backstage/backstage/blob/master/plugins/stack-overflow-backend/src/search/StackOverflowQuestionsCollatorFactory.ts#L82

Reply:
done.